Mathlib/Topology/Connected/Clopen.lean

```diff
@@ -563,3 +563,16 @@ theorem preconnectedSpace_of_forall_constant
     (hs : ∀ f : α → Bool, Continuous f → ∀ x y, f x = f y) : PreconnectedSpace α :=
   ⟨isPreconnected_of_forall_constant fun f hf x _ y _ =>
       hs f (continuous_iff_continuousOn_univ.mpr hf) x y⟩
+
+theorem preconnectedSpace_iff_clopen :
+    PreconnectedSpace α ↔ ∀ s : Set α, IsClopen s → s = ∅ ∨ s = Set.univ :=
+  ⟨fun _ _ => isClopen_iff.mp, fun h ↦ preconnectedSpace_of_forall_constant fun f hf x y =>
+    (h (f ⁻¹' {true}) ((isClopen_discrete {true}).preimage hf)).elim
+      (fun h => (eq_false_of_ne_true fun hx => Set.preimage_singleton_eq_empty.mp h ⟨x, hx⟩).trans
+        (eq_false_of_ne_true fun hy => Set.preimage_singleton_eq_empty.mp h ⟨y, hy⟩).symm)
+      (by aesop)⟩
```

Comment:
Would you mind making the proof of the reverse direction a bit more readable? I think this is a bit clearer:
```suggestion
  ⟨fun _ _ => isClopen_iff.mp, fun h ↦ by
    refine  preconnectedSpace_of_forall_constant fun f hf x y ↦ ?_
    have : f ⁻¹' {false} = (f ⁻¹' {true})ᶜ := by
      simpa [-Bool.compl_singleton] using congr((f ⁻¹' $(false.compl_singleton))ᶜ)
    obtain (h | h) := h _ ((isClopen_discrete {true}).preimage hf)
    all_goals simp_all⟩
```